optuna/exceptions.py

```diff
@@ -79,6 +79,16 @@ class DuplicatedStudyError(OptunaError):
     pass
 
 
+class StopStudy(OptunaError):
+    """Exception for stopping a study, once a threshold is achieved.
```

Comment:
Please note that the criteria would not only be exceeding a threshold but for more general purposes. For example, the exception may be raised in `GridSampler` to stop the study after all grid points are evaluated. (See also https://github.com/optuna/optuna/pull/665#discussion_r361424127 for more details.)

Would you mind modifying the docstring to cover more general scenarios?

Reply:
Thank you, I extended the docstring to include more general use cases.